test/modules/post/test/file.rb

```diff
@@ -46,6 +47,42 @@ def setup
     super
   end
 
+  def test_dir
+    fs_sep = session.platform == 'windows' ? '\\' : '/'
+
+    it 'should test for directory existence' do
+      ret = false
+      [
+        'c:\\',
+        '/etc/',
+        '/tmp'
+      ].each do |path|
+        ret = true if directory?(path)
+      end
+
+      ret
+    end
+
+    it 'should create directories' do
+      mkdir(datastore['BaseDirectoryName'])
+      ret = directory?(datastore['BaseDirectoryName'])
+      ret &&= write_file([datastore['BaseDirectoryName'], 'file'].join(fs_sep), '')
+      ret &&= mkdir([datastore['BaseDirectoryName'], 'directory'].join(fs_sep))
+      ret &&= write_file([datastore['BaseDirectoryName'], 'directory', 'file'].join(fs_sep), '')
```

Comment:
Should we validate these are created? 👀 

I noticed there's special handling for sym links in the new PRs that you've created, is that something we want to add tests for here too? I imagine that's the scenario that new Meterpreters might be implemented inconsistently 😄

Reply:
The follow up test will ensure that both 'file' and 'directory' are in the output when the directory is enumerated. Only `directory/file` isn't checked in that case.

As for symlinks, while you're correct the Meterpreters do handle that particular case, there isn't any first-class support for dealing with symlinks in the Meterpreters so I was leaning towards no. If we wanted to, we'd need to detect if the platform supported it and shell out to `ln -s` when it's available.

Reply:
I'm thinking that's a test worth adding, otherwise future Meterpreters won't implement it consistently 😄 

Testing the platform to run cmd_exec with `ln -s` or `mklink` might work just fine for this scenario, similar to the `cmd_exec` tests? 👀 

https://github.com/rapid7/metasploit-framework/blob/d49d823bf5c28c4f6d3e6668b8ff170a6600a57c/test/modules/post/test/cmd_exec.rb#L21

Reply:
Alright added two more tests in cf67dc130787bfb03037be705163ac222fabc820. The first checks that when the target is a symbolic link, it will be deleted. Something to note here is that even if the target is a symbolic link to a file, it'll still be deleted. That's sort of undefined behavior though since I think the original intention was just to cleanly delete symbolic links that pointed to directories. The second check ensures that when the target directory is deleted that if it contains a symbolic link into another directory, that link **is not followed**. This enforces consistency with the undocumented behavior I observed from Python's [`shutil.rmtree`](https://docs.python.org/3/library/shutil.html#shutil.rmtree).